crates/bevy_pbr/src/render/mesh_view_types.wgsl

```diff
@@ -61,6 +61,31 @@ struct Lights {
     spot_light_shadowmap_offset: i32,
 };
 
+struct Fog {
+    mode: u32,
+    base_color: vec4<f32>,
+    directional_light_color: vec4<f32>,
+    directional_light_exponent: f32,
+    // `be` and `bi` are allocated differently depending on the fog mode
+    //
+    // For Linear Fog:
+    //     be.x = start, be.y = end
+    // For Exponential and ExponentialSquared Fog:
+    //     be.x = density
+    // For Atmospheric Fog:
+    //     be = per-channel extinction density
+    //     bi = per-channel inscattering density
+    be: vec3<f32>,
+    bi: vec3<f32>,
+}
```

Comment:
Reordering for better packing. May bring a slight GPU-side performance benefit perhaps.
```suggestion
struct Fog {
    base_color: vec4<f32>,
    directional_light_color: vec4<f32>,
    // `be` and `bi` are allocated differently depending on the fog mode
    //
    // For Linear Fog:
    //     be.x = start, be.y = end
    // For Exponential and ExponentialSquared Fog:
    //     be.x = density
    // For Atmospheric Fog:
    //     be = per-channel extinction density
    //     bi = per-channel inscattering density
    be: vec3<f32>,
    directional_light_exponent: f32,
    bi: vec3<f32>,
    mode: u32,
}
```